bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/client/RequestLogger.java

```diff
@@ -91,18 +91,28 @@ private void dump(final Request request) {
                 }
             });
             request.onResponseContent((theResponse, content) -> contentBuffer
-                    .append(reformatJson(getCharset(theResponse.getHeaders()).decode(content).toString())));
+                    .append(getCharset(theResponse.getHeaders()).decode(content).toString()));
             request.onResponseSuccess(theResponse -> {
                 if (contentBuffer.length() > 0) {
                     group.append("\n");
-                    group.append(contentBuffer);
+                    group.append(reformatJson(contentBuffer.toString()));
                 }
+
                 String dataToLog = group.toString();
-                logger.debug(dataToLog);
+                scambleAndLog(stringsToRemove, dataToLog);
             });
         }
     }
 
+    private void scambleAndLog(String[] stringsToRemove, String dataToLog) {
+        for (String stringToRemove : stringsToRemove) {
+            if (stringToRemove != null) {
+                dataToLog = dataToLog.replace(stringToRemove, "<HIDDEN>");
```

Comment:
Can you fix the compiler warning? E.g. by storing dataToLog into a local variable.